plotly/plotly/plotly.py

```diff
@@ -1271,6 +1343,40 @@ def validate_credentials(credentials):
         raise exceptions.PlotlyLocalCredentialsError()
 
 
+def add_share_key_to_url(response, **payload):
+    """
+    Update plot's url to include the secret key
+
+    payload keyword agruments:
+    un (string) -- username from credentials file
+    kwargs (dict) --
+        'world_readable': (bool) make the plot public/ private
+        'sharing': ('public' | 'private' | 'secret')
+                    set privacy for the url
+
+    """
+    domain = (get_session_config().get('plotly_domain') or
+              get_config()['plotly_domain'])
+    plot_id = response['url'].replace(domain + "/~"
+                                      + payload['un'] + "/", "")
+
+    url = _api_v2.api_url("files/") + payload['un'] + ":" + plot_id
+    plot_kwargs = json.loads(payload['kwargs'])
+    new_response = requests.patch(url,
+                                  headers=_api_v2.headers(),
+                                  data={"share_key_enabled":
+                                        "True",
+                                        "world_readable":
+                                        plot_kwargs['world_readable']})
+
+    _api_v2.response_handler(new_response)
+
+    new_response_data = json.loads(new_response.content)
+    response['url'] += '?share_key=' + new_response_data['share_key']
```

Comment:
yeah, good point. `urlparse` can get you the query strings: https://docs.python.org/2/library/urlparse.html. This might end up being a useful function for folks once we have a `plotly.files.get` (i.e. polity's `$ ls`), so probably worth making this foolproof.